docs/technical_documents/open_space_decider.md

```diff
@@ -0,0 +1,175 @@
+
+    ```cpp
+          void SetOriginFromADC(Frame *const frame, const hdmap::Path &nearby_path);
+    ```
+
+    ```cpp
+          void SetParkAndGoEndPose(Frame *const frame);
+    ```
+
+    ```cpp
+          bool GetParkAndGoBoundary(Frame *const frame, const hdmap::Path &nearby_path,
+                                std::vector<std::vector<common::math::Vec2d>>
+                                *const roi_parking_boundary);
+    ```
+2. By calling formulateboundaryconstraints function to gather vertice needed by warm start and distance approach, then transform vertices into the form of Ax > b;
+
+```cpp
+    bool FormulateBoundaryConstraints(
+        const std::vector<std::vector<common::math::Vec2d>> &roi_parking_boundary,
+        Frame *const frame);
+```
+3. Return process status;
+
+4. Output: open space roi boundary and boundary constraints 
+
+## Open space pre stop decider
+
+1. Input: obstacles info \ vehicle info \ road info \ parking space info
+
+  - IN PARKING STAGE (OpenSpacePreStopDeciderConfig::PARKING)
+    1. Check parking space info(parking spot id and parking spot points), fill those info into target_parking_spot_ptr; take the s info of parking space center line as target_s;
+
+    ```cpp
+        bool CheckParkingSpotPreStop(Frame* const frame,
+                                    ReferenceLineInfo* const reference_line_info,
+                                    double* target_s);
+    ```
+    2. Base on ADC position, stop distance to target parking space and obstacle positon to set stop fence
+      
+    ```cpp
+        void SetParkingSpotStopFence(const double target_s, Frame* const frame,
+                                    ReferenceLineInfo* const reference_line_info);
+    ```
+  - IN PULL OVER STAGE (OpenSpacePreStopDeciderConfig::PULL_OVER)
+    1. Same with parking stage, check pull over pre stop -> set pull over stop fence
+
+    ```cpp
+        bool CheckPullOverPreStop(Frame* const frame,
+                                  ReferenceLineInfo* const reference_line_info,
+                                  double* target_s);
+    ```
+
+    ```cpp
+        void SetPullOverStopFence(const double target_s, Frame* const frame,
+                                ReferenceLineInfo* const reference_line_info);
+    ```
+2. Return process status
+
+3. Output: pre stop fence for open space planner
+
+## Open space fallback decider 
+1. Input: obstacles info \ vehicle info \ road info \ parking space info
+
+2. Based on the prdicted trajectory of obstacles, the bounding box info of obstacles in each time interval is obtained,then add into predicted_bounding_rectangles
+```cpp
+    void BuildPredictedEnvironment(const std::vector<const Obstacle*>& obstacles,
+                                   std::vector<std::vector<common::math::Box2d>>&
+                                   predicted_bounding_rectangles);
+```
+3. Calling IsCollisonFreeTrajectory function to determine whether it will intersect with obstacles;
```

Comment:
Personally, it would be nicer if IsCollisonFreeTrajectory is formatted as `IsCollisonFreeTrajectory()`.

Reply:
Since the original form is frequently used in this doc, why use Present Participle "Call-ing" here?

Reply:
Since the original form is frequently used in this doc, why use Present Participle "Call-ing" here?